core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/XMLReaderBatchSource.java

```diff
@@ -0,0 +1,304 @@
+          }
+        }
+      }
+    } catch (IOException exception) {
+      LOG.error("IOException occurred while reading temp directory path : " + exception.getMessage());
+    }
+  }
+
+  /**
+   * Config class that contains all the properties needed for the XML Reader.
+   */
+  public static class XMLReaderConfig extends ReferencePluginConfig {
+    @Description("Path to file(s) to be read. If a directory is specified, terminate the path name with a \'/\'.")
+    private final String path;
+
+    @Nullable
+    @Description("Pattern to select specific file(s)." +
+      "Example - " +
+      "1. Use '^' to select file with name start with 'catalog', like '^catalog'." +
+      "2. Use '$' to select file with name end with 'catalog.xml', like 'catalog.xml$'." +
+      "3. Use '*' to select file with name contains 'catalogBook', like 'catalogBook*'.")
+    private final String pattern;
+
+    @Description("Node path to emit individual event from the schema. " +
+      "Example - '/book/price' to read only price under the book node")
+    private final String nodePath;
+
+    @Description("Action to be taken after processing of the XML file. " +
+      "Possible actions are - " +
+      "1. Delete from the HDFS." +
+      "2. Archived to the target location." +
+      "3. Moved to the target location.")
+    private final String actionAfterProcess;
+
+    @Nullable
+    @Description("Target folder path if user select action after process, either ARCHIVE or MOVE. " +
+      "Target folder must be an existing directory.")
+    private final String targetFolder;
+
+    @Description("Name of the table to keep track of processed file(s).")
+    private final String reprocessingRequired;
+
+    @Description("Name of the table to keep track of processed file(s).")
+    private final String tableName;
+
+    @Description("Expiry period (days) for data in the table. Default is 30 days." +
+      "Example - For tableExpiryPeriod = 30, data before 30 days get deleted from the table.")
+    private final String tableExpiryPeriod;
+
+    @VisibleForTesting
+    XMLReaderConfig(String referenceName, String path, @Nullable String pattern,
+                           @Nullable String nodePath, @Nullable String actionAfterProcess,
+                           @Nullable String targetFolder, String reprocessingRequired, String tableName,
+                           String tableExpiryPeriod) {
+      super(referenceName);
+      this.path = path;
+      this.pattern = pattern;
+      this.nodePath = nodePath;
+      this.actionAfterProcess = actionAfterProcess;
+      this.targetFolder = targetFolder;
+      this.reprocessingRequired = reprocessingRequired;
+      this.tableName = tableName;
+      this.tableExpiryPeriod = tableExpiryPeriod;
+    }
+
+    @VisibleForTesting
+    String getTableName() {
+      return tableName;
+    }
+
+    @VisibleForTesting
+    String getReprocessingRequired() {
```

Comment:
I think it would be much better for this config to accept "true" or "false". Then, this method would return boolean, and line 274 would be ``Boolean.getBoolean(reprocessingRequired)``. That way, you don't have to do string comparisons everywhere.